students/16th/team2/konhee-kim/user/views.py

```diff
@@ -1 +1,129 @@
+import json
+import re
 
+from django.http  import JsonResponse
+from django.views import View
+
+from user.models  import User
+
+
+class SignUpView(View):
+
+    def post(self, request):
+        data          = json.loads(request.body)
+        email, mobile = check_email_or_mobile(data['email_or_mobile'])
+        username      = data['username']
+
+        if not (email or mobile): # unexpected value input
+            return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+        
+        if not(validate_password(data['password'])):
+            return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+
+        if not username:
+            return JsonResponse({'MESSAGE': 'KEY_ERROR'}, status=400)
+
+        # existence check
+        if email    and User.objects.filter(email         = email).exists():
+            return JsonResponse({'MESSAGE': 'EXISTENCE_ERROR'}, status=400)
+       
+        if mobile   and User.objects.filter(mobile_number = mobile).exists():
+            return JsonResponse({'MESSAGE': 'EXISTENCE_ERROR'}, status=400)
+        
+        if username and User.objects.filter(username      = username).exists():
+            return JsonResponse({'MESSAGE': 'EXISTENCE_ERROR'}, status=400)
```

Comment:
`and` 와 `=` 을 기준으로 정렬해주시려는 시도는 너무 좋았지만 여기서는 오히려 가독성이 좋지 못합니다. 아래처럼 수정해주세요. 그리고 `exists()` 사용 너무 good 입니다 👍
```suggestion
        if email and User.objects.filter(email = email).exists():
            return JsonResponse({'MESSAGE': 'EXISTENCE_ERROR'}, status=400)
       
        if mobile and User.objects.filter(mobile_number = mobile).exists():
            return JsonResponse({'MESSAGE': 'EXISTENCE_ERROR'}, status=400)
        
        if username and User.objects.filter(username = username).exists():
            return JsonResponse({'MESSAGE': 'EXISTENCE_ERROR'}, status=400)
```